### PR TITLE
Logging improvements

### PR DESCRIPTION
**Pull request by stohrendorf**
* removes console from release versions unless there's no location to write a log file to
* preserves the last 10 log files and copies them over to the bug reports
* adjustment to the readme to account for the new bug report action

### CODE REVIEW

**Review by serishema — CHANGES_REQUESTED** [review]
I tested this and the log rotation and bug report works.

But the console window still appears in release builds. 
This is a bit of a pain to do portably:
To get rid of it on windows you have to change main() to WinMain() (see MSDN) when compiling for windows 
and then in cmake change
`add_executable( croftengine ${CROFTENGINE_SRCS} )`
to
`add_executable( croftengine WIN32 ${CROFTENGINE_SRCS} )`
you will probably need to make that conditional in cmake so it only does it when building for windows, i couldn't find whether the WIN32 directive is silently ignored when compiling for other platforms or not.
Then when you need a console window you can call AllocConsole() to create one.

**Comment by stohrendorf** [comment]
Good catch, somehow my IDE interferred with the console sometimes. About `WIN32` in `add_executable`, I tested this through WSL, and it does nothing on Linux, because it only adds a property to the executable that's simply ignored on non-windows platforms.

File: src/croftengine.cpp [hunk 1]
```diff
@@ -45,26 +49,54 @@ void terminateHandler()
...
+const gsl::czstring logFormat = "[%TimeStamp% %Severity% %ThreadID%] %Message%";
+
+void initConsole(boost::log::trivial::severity_level level)
+{
+#ifdef WIN32
+  AllocConsole();
+
+  // https://stackoverflow.com/a/57241985
+  FILE* fDummy;
+  freopen_s(&fDummy, "CONIN$", "r", stdin);
```
Comment (serishema) [inline src/croftengine.cpp:62]: I forgot about reopening the standard input/output streams.

**Review by serishema — APPROVED** [review]
Good. This works correctly now.

**Pull request merged**